book/command_signature.md

```diff
@@ -1,24 +1,22 @@
 # Command signature
 
-nu commands contain a signature section, take [`str distance`](/commands/docs/str_distance.md) as example, the signature is like this:
+nu commands can be given explicit signatures; take [`str distance`](/commands/docs/str_distance.md) as an example, the signature is like this:
 
 ```nu
-Signatures(Cell paths are supported):
-  <string> | str distance <string> -> <int>
+def "str distance" [compare-string: string, ...rest: cell-path]: string -> int { }
 ```
 
-The first type name before `|` describes the type of input pipeline. The command name is followed by the required argument type(s) for the command. The output type is `int` and given after `->`.
+The type names after after the `:` describe the command's input/output pipeline types. The input type for a given pipeline, in this case `string`, is given before the `->`; and the output type `int` is given after `->`. There can be multiple input/output types. If there are multiple input/output types, they can be placed within brackets and separated with commas: `str distance`'s signature actually has the `[string -> int, table -> table, record -> record]` pipeline types.
 
-`(Cell paths are supported)` indicates that you can provide cell paths for `str distance` to apply an operation at the given cell path(s) in a nested structure or table, and replace the column or field with the result, like:  `ls | str distance 'nushell' 'name'`
+The `cell-path` type indicates that you can provide cell paths for `str distance` to apply an operation at the given cell path(s) in a nested structure or table, and replace the column or field with the result, like: `ls | str distance 'nushell' 'name'`
 
-Here is another one example, [`str join`](/commands/docs/str_join.md):
+Here is another example, [`str join`](/commands/docs/str_join.md):
 
 ```nu
-Signatures:
-  list<string> | str join <string?> -> <string>
+def "str join" [separator?: string]: [list<string> -> string, string -> string] { }
```

Comment:
```suggestion
def "str join" [separator?: string]: [list<any> -> string, string -> string] { }
```

Reply:
I replaced `list<string>` with `list` in `42ea6a8` to clarify in text that it implies `list<any>`.